app/controllers/general_shopping_list_controller.rb

```diff
@@ -0,0 +1,12 @@
+class GeneralShoppingListController < ApplicationController
+  def index
+    recipe_foods = RecipeFood.pluck(:food_id)
+    foods = Food.all
+    @foods = []
+    foods.each do |food|
+      @foods.push(food) unless food.in?(recipe_foods)
+    end
+    prices = @foods.map { |n| n.price * n.quantity }
+    @prices = prices.sum
+  end
+end
```

Comment:
- [ ] You have done a good job trying to implement the general shopping list functionality. However, at the moment, it basically only displays all the food items in the database. Please look into it, because the general shopping list should actually generate food to buy based on the missing or inadequate food for your recipes 👍 

**NOTE:** A big issue that I'd like to address, and I believe is part of this issue, is that it is impossible to add ingredients to the recipes in your app at the moment. And that is very important for the shopping list to even be accurate.